cmd/xl-storage.go

```diff
@@ -229,14 +229,23 @@ func makeFormatErasureMetaVolumes(disk StorageAPI) error {
 
 // Initialize a new storage disk.
 func newXLStorage(ep Endpoint, cleanUp bool) (s *xlStorage, err error) {
-	path := ep.Path
-	if path, err = getValidPath(path); err != nil {
-		return nil, err
+	s = &xlStorage{
+		drivePath:  ep.Path,
+		endpoint:   ep,
+		globalSync: globalFSOSync,
+		poolIndex:  -1,
+		setIndex:   -1,
+		diskIndex:  -1,
```

Comment:
We make sure to never return `nil` for `s *xlStorage` - we will always return `error` if any however the disk itself always is an initialized data structure.